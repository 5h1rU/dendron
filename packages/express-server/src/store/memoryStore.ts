import _ from "lodash";

const STORE: any = {};

export class MemoryStore {
  static _instance: MemoryStore;

  static instance() {
    if (!MemoryStore._instance) {
      MemoryStore._instance = new MemoryStore();
    }
    return MemoryStore._instance;
  }

  async put(key: string, value: any) {
    STORE[key] = value;
  }

  async get<T>(key: string): Promise<T | undefined> {
    return STORE[key] as T;
  }

  async list(prefix: string): Promise<any> {
    return _.filter(STORE, (ent: string) => ent.startsWith(prefix));
  }
}
