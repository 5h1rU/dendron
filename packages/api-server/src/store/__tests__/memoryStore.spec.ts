import { MemoryStore } from "../memoryStore";

describe("memory store", () => {
  let store: MemoryStore;

  beforeEach(() => {
    store = MemoryStore.instance(true);
  });

  test("basic", async () => {
    await store.put("foo", 1);
    const resp = await store.list("foo");
    expect(resp).toMatchSnapshot();
  });
});
