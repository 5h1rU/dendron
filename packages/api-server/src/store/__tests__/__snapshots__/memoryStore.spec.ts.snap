// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`memory store basic 1`] = `
Object {
  "foo": 1,
}
`;
